src/pykeen/datasets/base.py

```diff
@@ -611,7 +610,7 @@ class TarFileSingleDataset(LazyDataset):
     def __init__(
         self,
         url: str,
-        relative_path: str,
+        relative_path: pathlib.PurePosixPath,
```

Comment:
keep backwards compatibility - if it gets a str, convert to a PurePosixPath

Reply:
Why? This is an internal abstraction layer, isn't it?

Reply:
I think we should make sure this is still usable by someone who wants to create an ad-hoc dataset